scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -274,6 +274,83 @@ def run_build(self, args=[]):
 
         return results
 
+    def run_filter_cmake(self, args=""):
+
+        if not self.options.disable_warnings_as_errors:
```

Comment:
You are duplicating alot of code from run_cmake. Please figure out how to define and do this only in one place

Reply:
@nashif I reworked the PR removing duplicated methods and reusing the existing ones. It ended up with only a small expansion needed